backend/components/installer/src/main/java/co/airy/core/components/installer/CatalogHandler.java

```diff
@@ -0,0 +1,179 @@
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Service;
+
+import co.airy.core.api.components.installer.HelmJobHandler;
+import co.airy.core.api.components.installer.NotCompletedException;
+import co.airy.core.api.components.installer.model.ComponentDetails;
+import co.airy.log.AiryLoggerFactory;
+
+@Service
+public class CatalogHandler implements ApplicationListener<ApplicationReadyEvent>, DisposableBean {
+
+    private static final Logger log = AiryLoggerFactory.getLogger(CatalogHandler.class);
+
+    private final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    private final File repoFolder;
+    private final String catalogUri;
+    private final ApiClient apiClient;
+    private final String namespace;
+    private final HelmJobHandler helmJobHandler;
+    private Git git;
+
+    CatalogHandler(
+            ApiClient apiClient,
+            HelmJobHandler helmJobHandler,
+            @Value("${catalog.uri}") String catalogUri,
+            @Value("${catalog.directory}") String catalogDir,
+            @Value("${kubernetes.namespace}") String namespace) {
+        this.apiClient = apiClient;
+        this.catalogUri = catalogUri;
+        this.repoFolder = new File(catalogDir);
+        this.helmJobHandler = helmJobHandler;
+        this.namespace = namespace;
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        try {
+            git = Git.cloneRepository()
+                .setURI(catalogUri)
+                .setDirectory(repoFolder)
+                .call();
+        } catch (GitAPIException e) {
+            throw new IllegalStateException("not able to clone catalog repository", e);
+        }
+    }
+
+    @Override
+    public void destroy() {
+        if (repoFolder != null) {
+            repoFolder.delete();
+        }
+    }
+
+    public List<ComponentDetails> listComponents() throws Exception {
+        return getComponents((s -> true));
+    }
+
+    public ComponentDetails getComponentByName(String componentName) throws Exception {
+        return getComponents((s -> s.equals(componentName)))
+            .stream()
+            .findAny()
+            //FIXME: throws exception instead
+            .orElse(null);
+    }
+
+    //TODO: Add return value and correct exception handleling and return value
+    private List<ComponentDetails> getComponents(Function<String, Boolean> condition) throws Exception {
+        git.pull();
```

Comment:
curious: What's the reason for using this over the GitHub tree API you mentioned?